controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1122,6 +1124,177 @@ bool master_thread::handle_cmdu_1905_channel_selection_response(Socket *sd,
+            for (unsigned int i = 0; i < vrx.size(); ++i) {
+                LOG(DEBUG)
+                    << "rx interface metric data # " << i << "  neighbor al_mac:"
+                    << network_utils::mac_to_string(vrx[i].mac_of_an_interface_in_the_neighbor_al)
+                    << "  receiving al_mac:"
+                    << network_utils::mac_to_string(vrx[i].mac_of_an_interface_in_the_receiving_al)
+                    << "  rssi= " << std::hex << int(vrx[i].link_metric_info.rssi_db) << std::endl;
+            }
+
+            auto &vtx = pair_neighbor.second.transmitterLinkMetrics;
+            for (unsigned int i = 0; i < vtx.size(); i++) {
+                LOG(DEBUG)
+                    << "tx interface metric data # " << i << "  neighbor al_mac:"
+                    << network_utils::mac_to_string(vtx[i].mac_of_an_interface_in_the_neighbor_al)
+                    << "  receiving al_mac:"
+                    << network_utils::mac_to_string(vtx[i].mac_of_an_interface_in_the_receiving_al)
+                    << "  phy_rate= " << std::hex << int(vtx[i].link_metric_info.phy_rate)
+                    << std::endl;
+            }
+        }
+    }
+}
+
+bool master_thread::handle_cmdu_1905_link_metric_response(Socket *sd,
+                                                          ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+
+    auto mid = cmdu_rx.getMessageId();
+    LOG(INFO) << "Received LINK_METRIC_RESPONSE_MESSAGE, mid=" << std::dec << int(mid);
+
+    //getting reference for link metric data storage from db
+    auto &m_link_metric_data = database.get_link_metric_data_map(database.get_gw_mac());
+
+    //will hold new link metric data from Reporting Agent
+    son::node::link_metrics_data new_link_metric_data;
+
+    int tlvType;
+    // holds mac address of the reporting agent, used as m_link_metric_data key
+    sMacAddr reporting_agent_al_mac, neighbor_al_mac;
+    reporting_agent_al_mac.struct_init();
+    neighbor_al_mac.struct_init();
+
+    // parse all tlvs in cmdu and save ap metric data to database
+    while ((tlvType = cmdu_rx.getNextTlvType()) != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE)) {
+
+        LOG(DEBUG) << "getNextTlvType";
+        if (tlvType < 0) {
+            LOG(ERROR) << "getNextTlvType has failed";
+            return false;
+        }
+
+        if (tlvType == int(ieee1905_1::eTlvType::TLV_TRANSMITTER_LINK_METRIC)) {
+            // parse tx_Link_metric_data
+            auto TxLinkMetricData = cmdu_rx.addClass<ieee1905_1::tlvTransmitterLinkMetric>();
+            if (!TxLinkMetricData) {
+                LOG(ERROR) << "addClass ieee1905_1::tx_Link_metric_data has failed";
+                return false;
+            }
+
+            reporting_agent_al_mac = TxLinkMetricData->al_mac_of_the_device_that_transmits();
+            neighbor_al_mac =
+                TxLinkMetricData
+                    ->al_mac_of_the_neighbor_whose_link_metric_is_reported_in_this_tlv();
```

Comment:
please rename this variable name in yaml file. It is not reasonable to use such a long name.